### PR TITLE
Add Pushsafer.com notification service

### PR DESCRIPTION
**Pull request by appzer**
Pushsafer
**Add pushsafer.com notification service**
> tested in live environment
> send notifications to iOS, Android or Windows 10 devices with chart image (last 20 uptime records)

Example Screenshots:
**iOS**
Pushsafer
Pushsafer
**Android**
Pushsafer
**WIN 10 Phone**
Pushsafer
**WIN 10 Desktop**
Pushsafer
**Client Apps**
Pushsafer

### CODE REVIEW

**Comment by naqirizvi** [comment]
Also looking for some integration with OneSignal.com

**Comment by appzer** [comment]
i fixed the issues

**Comment by appzer** [comment]
How long it takes to review the pr

**Comment by sadortun** [comment]
@appzer sorry for the delay,  we did not had time yet to review and test your PR

**Comment by appzer** [comment]
hi, where the problem is? In my development environment, all works fine.

Kevin

File: src/psm/Module/Config/Controller/ConfigController.php [hunk 1]
```diff
@@ -66,6 +68,13 @@ class ConfigController extends AbstractController {
 		'sms_gateway_password',
 		'sms_from',
 		'pushover_api_token',
+		'pushsafer_icon_on',
```
Comment (TimZ99) [inline src/psm/Module/Config/Controller/ConfigController.php:71]: Is it really necessary to have _on and _off? 
Can't this be 1 variable?

File: src/psm/Util/Server/Updater/StatusNotifier.php [hunk 2]
```diff
@@ -279,6 +292,160 @@ protected function notifyByPushover($users) {
 	}
 
 	/**
+	 * This functions performs the pushsafer chart generation
+	 */
+	protected function PushsaferChart($values) {
```
Comment (TimZ99) [inline src/psm/Util/Server/Updater/StatusNotifier.php:297]: Maybe move this function to functions.inc.php?

File: src/psm/Util/Install/Installer.php [hunk 3]
```diff
@@ -150,12 +150,21 @@ public function install() {
 					('sms_from', '1234567890'),
 					('pushover_status', '0'),
 					('pushover_api_token', ''),
+					('pushsafer_status', '0'),
```
Comment (TimZ99) [inline src/psm/Util/Install/Installer.php:153]: Same as src/psm/Module/Config/Controller/ConfigController.php, why _on and _off, why not one variable?

File: src/templates/default/module/config/config.tpl.html [hunk 4]
```diff
@@ -265,6 +266,73 @@
...
+			<div class="control-group">
+                <label class="control-label" for="pushsafer_sound_off">{{ label_pushsafer_sound }} <span class="label label-status-off">off</span></label>
+                <div class="controls">
+                    <input type="text" id="pushsafer_sound_off" name="pushsafer_sound_off" value="{{ pushsafer_sound_off }}" maxlength="3" />
+                </div>
+            </div>
+			<div class="control-group">
+                <label class="control-label" for="pushsafer_vibration_on">{{ label_pushsafer_vibration }} <span class="label label-status-on">on</span></label>
+                <div class="controls">
+                    <input type="text" id="pushsafer_vibration_on" name="pushsafer_vibration_on" value="{{ pushsafer_vibration_on }}" maxlength="3" />
```
Comment (TimZ99) [inline src/templates/default/module/config/config.tpl.html:310]: What does an input look like? yes/no, true/false, a number or ... ? 
Please use a placeholder or add some text.

File: src/templates/default/module/server/server/list.tpl.html [hunk 5]
```diff
@@ -23,7 +23,7 @@
 			<td class="nowrap visible-desktop"><div class="table-cell-title">{{ server.type }}</div></td>
 			<td class="nowrap hidden-phone"><div class="table-cell-title">{{ server.rtime }} s</div></td>
 			<td class="nowrap hidden-phone"><div class="table-cell-title">{{ server.last_online }}</div></td>
-			<td class="tight hidden-phone"><div class="table-cell-title"><i class="{{ server.active_icon }}" title="{{ server.active_title }}"></i> <i class="{{ server.email_icon }}" title="{{ label_email }}"></i> <i class="{{ server.sms_icon }}" title="{{ label_sms }}"></i> <i class="{{ server.pushover_icon }}" title="Pushover"></i></div></td>
+			<td class="tight hidden-phone"><div class="table-cell-title"><i class="{{ server.active_icon }}" title="{{ server.active_title }}"></i> <i class="{{ server.email_icon }}" title="{{ label_email }}"></i> <i class="{{ server.sms_icon }}" title="{{ label_sms }}"></i> <i class="{{ server.pushover_icon }}" title="Pushover"></i> <i class="{{ server.pushsafer_icon }}" title="Pushsafer"></i></div></td>
```
Comment (TimZ99) [inline src/templates/default/module/server/server/list.tpl.html:26]: `title="Pushsafer"` must be `title="{{ label_pushsafer }}"` to be inline with the rest.
Same for `title="Pushsafer"` must be `title="{{ label_pushover }}"`

File: static/css/style.css [hunk 6]
```diff
@@ -1,674 +1,575 @@
...
-	background-color: #f5f5f5;
-	-webkit-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.05);
-	-moz-box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.05);
-	box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.05);
-}
-.nav-tabs > li > a, .nav-tabs > li > a:hover {
-	padding: 8px 8px;
-}
-
-/** Rtl language Style */
```
Comment (TimZ99) [inline static/css/style.css:117]: Why is the Rtl language part gone?

File: src/psm/Util/Server/Updater/StatusNotifier.php [hunk 7]
```diff
@@ -279,6 +292,160 @@ protected function notifyByPushover($users) {
...
+			foreach ($records as $uptime) {
+				$chartvalues[$uptime['date']] = $uptime['latency'];
+			}
+			$picture = $this->PushsaferChart($chartvalues);
+		}
+		
+	    foreach($users as $user) {
+			$userlist[] = $user['user_id'];
+			if(empty($user['pushsafer_key'])) {
+				$this->addMessage(psm_get_lang('config', 'pushsafer_error_nokey'), 'error');
```
Comment (TimZ99) [inline src/psm/Util/Server/Updater/StatusNotifier.php:408]: When the server is offline and you click update you get: `Fatal error: Call to undefined method psm\Util\Server\Updater\StatusNotifier::addMessage() in /src/psm/Util/Server/Updater/StatusNotifier.php on line 408`

**Comment by TimZ99** [comment]
Are the `$#39;` you see around the server name in the IOS screenshot also in this version, if yes please fix, if no ignore this message 😉.

**Review by TimZ99 — CHANGES_REQUESTED** [review]
On a fresh install non of the server and profile settings will save. 
Saving config tab works fine. Don't know why.

When I add a server and add the Pushsafer key, manually via the database, I get a Fatal error when updating (via navigation menu). 

Not ready for production, this version isn't working for me.

**Comment by TimZ99** [comment]
Hi, I know it's a long time ago you made the PR, but I'm currently reviewing the open PR's and it seams that this one is not ready to be merged. Could you please check the review and fix the conflicts. :smile:

**Pull request closed without merging**